scripts/CMSRunAnalysis.py

```diff
@@ -176,39 +185,42 @@ def logCMSSW():
     keepAtEnd   = 3000
     maxLineLen  = 3000
     maxLines    = keepAtStart + keepAtEnd
-    numLines = sum(1 for line in open(outfile))
+    with open(outfile, encoding='utf-8') as fh:
+        numLines = sum(1 for l in fh)
 
     print("======== CMSSW OUTPUT STARTING ========")
-    print("NOTICE: lines longer than %s characters will be truncated" % maxLineLen)
+    print(f"NOTICE: lines longer than {maxLineLen} characters will be truncated")
 
     tooBig = numLines > maxLines
     prefix = "== CMSSW: "
     if tooBig:
-        print("WARNING: output more than %d lines; truncating to first %d and last %d" % (maxLines, keepAtStart, keepAtEnd))
+        print(f"WARNING: output more than {maxLines} lines; truncating to first {keepAtStart} and last {keepAtEnd}")
         print("Use 'crab getlog' to retrieve full output of this job from storage.")
         print("=======================================")
-        with open(outfile) as fp:
-            for nl, line in enumerate(fp):
+        with open(outfile, encoding='utf-8') as fp:
+            for nl, aLine in enumerate(fp):
                 if nl < keepAtStart:
-                    printCMSSWLine("== CMSSW: %s " % line, maxLineLen)
+                    printCMSSWLine(f"== CMSSW: {aLine} ", maxLineLen)
                 if nl == keepAtStart + 1:
                     print(prefix)
                     print(prefix + " [...BIG SNIP...]")
                     print(prefix)
                 if numLines - nl <= keepAtEnd:
-                    printCMSSWLine("%s" % prefix+line, maxLineLen)
+                    printCMSSWLine(f"{prefix + aLine}", maxLineLen)
     else:
-        for line in open(outfile):
-            printCMSSWLine("%s" % prefix+line, maxLineLen)
+        with open(outfile, encoding='utf-8') as fh:
+            for aLine in fh:
+                printCMSSWLine(f"{prefix + aLine}", maxLineLen)
 
     print("======== CMSSW OUTPUT FINSHING ========")
     logCMSSWSaved = True
-    open('logCMSSWSaved.txt', 'a').close()
+    with open('logCMSSWSaved.txt', 'a', encoding='utf-8') as fh:
+        fh.close()
     os.utime('logCMSSWSaved.txt', None)
```

Comment:
Same comment as above

Reply:
changed open as indicated and removed call to `os.utime`